autoload/mr/recorder.vim

```diff
@@ -40,6 +42,24 @@ function! s:recorder_record(filename) abort dict
   call s:dump_delay(self)
 endfunction
 
+function! s:recorder_delete(...) abort dict
+  let list = a:000
+  if a:0 == 1 && type(a:1) == 3
+   " accept a list
+    let list = a:1
+  endif
+  if len(list) == 0
+    return
+  endif
+  let self._deletes = reduce(
+        \ list,
+        \ {acc, v -> 
+            \ extend(acc, {simplify(resolve(fnamemodify(v, ':p'))): 1})
+        \ },
+        \ self._deletes)
+  call s:dump_delay(self)
+endfunction
```

Comment:
I don't think there is enough advantage to make it complicated for supporting string and list in `mr#mrX#delete()` function.

So please rewrite the function to support only string like

```vim
function! s:recorder_delete(filename) abort dict
  let norm = simplify(resolve(fnamemodify(a:filename, ':p')))
  let self._deletes[norm] = 1
  call s:dump_delay(self)
endfunction
```

Or only list like

```vim
function! s:recorder_delete(filenames) abort dict
  let self._deletes = reduce(
        \ a:filenames,
        \ { acc, v -> extend(acc, {simplify(resolve(fnamemodify(v, ':p'))): 1}) },
        \ self._deletes,
        \)
  call s:dump_delay(self)
endfunction
```

I think the function is called from users' explicit actions so handy is more important than performance or flexibility.
So please choose `mr#mrX#delete({string})` more than `mr#mrX#delete({list})`, if you don't have strong favor.

Reply:
I agree with you and fixed it: 2a11c28